lib/versioncontrol/upgradewindow/upgradewindow.go

```diff
@@ -427,6 +477,18 @@ func (e *systemdDriver) Sync(ctx context.Context, rsp proto.ExportUpgradeWindows
 	return nil
 }
 
+func (e *systemdDriver) SyncAuthVersion(ctx context.Context, version string) error {
+	if err := os.MkdirAll(e.cfg.ConfigDir, defaults.DirectoryPermissions); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := os.WriteFile(e.versionFile(), []byte(version), defaults.FilePermissions); err != nil {
```

Comment:
@fspmarshall shouldn't this (and the schedule one) be atomic writes? What happens if the upgrader runs right as we're writing the file?